src/Raven.Server/Documents/DocumentDatabase.cs

```diff
@@ -1413,7 +1421,7 @@ private async ValueTask NotifyFeaturesAboutStateChangeAsync(DatabaseRecord recor
 
             while (taken == false)
             {
-                taken = await _clusterLocker.WaitAsync(TimeSpan.FromSeconds(5));
+                taken = await _updateDatabaseRecordLocker.WaitAsync(TimeSpan.FromSeconds(5));
```

Comment:
this can throw object disposed ex, are we taking that into account?

Reply:
Wondering if we should dispose them or leave them and let the DatabaseShutdown.IsCancellationRequested act few lines below?

Reply:
This is handled here:
https://github.com/ravendb/ravendb/blob/2955874a713a26a8a57b787c17bdd1be0a4cb4dc/src/Raven.Server/Documents/DatabasesLandlord.cs#L206

Reply:
After looking at this again, we exit if the database is shutting down:
https://github.com/ravendb/ravendb/blob/2955874a713a26a8a57b787c17bdd1be0a4cb4dc/src/Raven.Server/Documents/DocumentDatabase.cs#L1423
And after this we notify the listeners that the command was completed successfully.
https://github.com/ravendb/ravendb/blob/2955874a713a26a8a57b787c17bdd1be0a4cb4dc/src/Raven.Server/Documents/DocumentDatabase.cs#L1383

Which isn't correct.